### PR TITLE
test(cluster-migration): Fix some bugs and add cluster migration fuzzy tests

### PR DESCRIPTION
**Pull request by chakaz**

### CODE REVIEW

File: src/server/cluster/cluster_family.cc [hunk 1]
```diff
@@ -896,7 +895,9 @@ void ClusterFamily::DflyMigrateFullSyncCut(CmdArgList args, ConnectionContext* c
   std::lock_guard lck(migration_mu_);
   auto migration_it =
       std::find_if(incoming_migrations_jobs_.begin(), incoming_migrations_jobs_.end(),
-                   [sync_id = sync_id](const auto& el) { return el->GetSyncId() == sync_id; });
+                   [sync_id = sync_id, cntx](const auto& el) {
```
Comment (BorysTheDev) [inline src/server/cluster/cluster_family.cc:898]: you don't need sync_id

File: src/server/cluster/cluster_slot_migration.h [hunk 2]
```diff
@@ -31,6 +31,12 @@ class ClusterSlotMigration : ProtocolClient {
     return sync_id_;
   }
 
+  // Remote sync ids uniquely identifies a sync *remotely*. However, multiple remote sources can
+  // use the same id, so we need a local id as well.
+  uint32_t GetLocalSyncId() const {
```
Comment (BorysTheDev) [inline src/server/cluster/cluster_slot_migration.h:36]: good catch

**Review by BorysTheDev — APPROVED** [review]

File: tests/dragonfly/cluster_test.py [hunk 3]
```diff
@@ -953,3 +953,178 @@ async def test_cluster_data_migration(df_local_factory: DflyInstanceFactory):
...
+    instance: DflyInstance
+    client: aioredis.Redis
+    admin_client: aioredis.Redis
+    slots: list
+    next_slots: list
+    sync_ids: list
+
+
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_cluster_fuzzymigration(df_local_factory: DflyInstanceFactory, df_seeder_factory):
```
Comment (adiholden) [inline tests/dragonfly/cluster_test.py:972]: please take some logic outside to utility functions

File: tests/dragonfly/cluster_test.py [hunk 4]
```diff
@@ -953,3 +953,178 @@ async def test_cluster_data_migration(df_local_factory: DflyInstanceFactory):
...
+    node_count = 3
```
Comment (dranikpg) [inline tests/dragonfly/cluster_test.py:973]: we should make it configurable later to start more nodes

**Pull request merged**